### PR TITLE
[WPB-8881] Add unit tests for new effect actions

### PR DESCRIPTION
**Pull request by mdimjasevic**
This is a follow-up PR to #4316 that adds unit tests for effect actions that were introduced in that PR.

Tracked by https://wearezeta.atlassian.net/browse/WPB-8881

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/wire-subsystems/test/unit/Wire/UserStoreSpec.hs [hunk 1]
```diff
@@ -33,3 +37,21 @@ spec = do
...
+  describe "UserStore effect" $ do
+    prop "user self email deleted" $ \user1 user2' email2 config ->
+      let user2 = user2' {email = Just email2}
+          localBackend = def {users = [user1, user2]}
+          result =
+            runNoFederationStack localBackend Nothing config $ do
+              deleteEmail (user1.id)
+              gets users
+       in result === [user1 {email = Nothing}, user2]
+    prop "update unvalited email" $ \user1 user2 email1 config ->
```
[the remark this answers is not in the document]
Comment (mdimjasevic) [reply libs/wire-subsystems/test/unit/Wire/UserStoreSpec.hs:50]: Uh, yes, you are right! I had such a typo elsewhere already before creating a PR, but haven't noticed it here! Thanks!

**Comment by mdimjasevic** [comment]
@supersven , is there anything else you'd like me to change, update, or do about it?

File: libs/wire-subsystems/test/unit/Wire/ActivationCodeStore/InterpreterSpec.hs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import Test.Hspec.QuickCheck
+import Test.QuickCheck
+import Wire.API.User.Activation
+import Wire.ActivationCodeStore
+import Wire.MiniBackend
+import Wire.MockInterpreters.ActivationCodeStore
+
+spec :: Spec
+spec = do
+  describe "ActivationCodeStore effect" $ do
```
Comment (supersven) [inline libs/wire-subsystems/test/unit/Wire/ActivationCodeStore/InterpreterSpec.hs:16]: Have you considered to add a test where the lookup fails? E.g. there is a code, but the key does not match?

File: libs/wire-subsystems/test/unit/Wire/UserStoreSpec.hs [hunk 3]
```diff
@@ -33,3 +37,21 @@ spec = do
        in if (isJust storedUser.language)
             then user.userLocale === Locale (fromJust storedUser.language) storedUser.country
             else user.userLocale === defaultLocale
+
+  describe "UserStore effect" $ do
```
Comment (supersven) [inline libs/wire-subsystems/test/unit/Wire/UserStoreSpec.hs:41]: I'm wondering if it wouldn't be useful to also test what happens when the user does not exist. Or, can't this case not happen? :thinking:

Comment (supersven) [reply libs/wire-subsystems/test/unit/Wire/UserStoreSpec.hs:41]: Ah, I see. Then, I'm approving. :+1:

**Review by supersven — APPROVED** [review]

File: libs/wire-subsystems/test/unit/Wire/ActivationCodeStore/InterpreterSpec.hs (hunk 2 above)
Comment (mdimjasevic) [reply libs/wire-subsystems/test/unit/Wire/ActivationCodeStore/InterpreterSpec.hs:16]: Just added it! I've rebased and pushed.

Comment (supersven) [reply libs/wire-subsystems/test/unit/Wire/ActivationCodeStore/InterpreterSpec.hs:16]: Cool, thanks! (The more test code coverage we can add, the better - as long as the task doesn't grow out of bounds, of course :smile: )

**Pull request merged**